frontend/projects/upgrade/src/app/shared/services/common-dialog.service.ts

```diff
@@ -260,9 +260,9 @@ export class DialogService {
     return this.openUpsertPrivateSegmentListModal(commonModalConfig);
   }
 
-  openAddListModal(appContext: string, segmentId: string) {
+  openAddListModal(appContext: string, segmentId: string, segmentType: SEGMENT_TYPE) {
```

Comment:
you can just pass in the title directly instead of conditionalizing based on type